content/inside-rust/this-development-cycle-in-cargo-1.92.md

```diff
@@ -0,0 +1,376 @@
+The concern is if a new field gets added,
+should we err on the side of it being supported or not?
+Forgetting to update the Cargo script allowlist on the release of a new feature is a poor experience.
+On the other hand, forgetting to update the denylist could mean we commit to a feature that we shouldn't support.
+The ideal solution is to rely on the type system to ensure we exhausitvely the manifest fields.
+If that isn't possible, we erred on the side of an allowlist.
+Thankfully, the implementation had already been updated to make it easy to rely on the type system for this.
+The validation logic was changed in
+[#16026](https://github.com/rust-lang/cargo/pull/16026).
+
+<!-- team meeting 2025-10-07 for escaping package.name -->
+A cargo script's file name gets turned into a
+[`package.name`](https://doc.rust-lang.org/cargo/reference/manifest.html#the-name-field)
+but not every script name is a valid `package.name`.
+So far, Cargo has sanitized the file name into being a valid `package.name`.
+But valid according to whom?
+General Cargo commands,
+`cargo new`,
+or crates.io?
+So far, the `cargo new` rules had been implemented.
+This is important to decide upfront because the sanitization results are visible through the binary's name, `cargo metadata`, and `--message-format json`.
+As we stepped through each `cargo new` rule,
+we found they were becoming less relevant through other efforts in Cargo, changes in Windows, etc.
+We decided to do the bare minimum sanitization needed for general Cargo commands.
+During the implementation of
+[#16120](https://github.com/rust-lang/cargo/pull/16120),
+[epage](https://github.com/epage/)
+felt it was too premature to freely allow names that match directory names inside the binary's directory.
+Users can just now move some of those directories out in Rust 1.91
+([#15833](https://github.com/rust-lang/cargo/pull/15833)).
+Changing this to be the default in Cargo is still under discussion
+([#16147](https://github.com/rust-lang/cargo/issues/16147))
+and users could still move it back.
+Instead of sanitizing,
+epage let this fall back to existing validation rules that will error.
+
+### Public dependencies
+
+*Update from [1.76](https://blog.rust-lang.org/inside-rust/2024/01/03/this-development-cycle-in-cargo-1-76/#rfc-3516-public-private-dependencies)*
+
+While this feature is largely blocked on the lint within rustc,
+this was further refined in Cargo.
+
+[jneem](https://github.com/jneem) experimented with Cargo rewriting the lint to provide Cargo-specific context in [#16002](https://github.com/rust-lang/cargo/pull/16002).
+
+[sadmac7000](https://github.com/sadmac7000) changed `cargo add`s version auto-selection to evaluate public dependencies in case the user intends to use them together
+([#15966](https://github.com/rust-lang/cargo/pull/15966)).
+
+<!-- team meeting 2025-10-07 -->
+[JohnScience](https://github.com/JohnScience) proposed `cargo tree --edges no-external` as a way to see only local packages
```

Comment:
I didn't even realized these efforts were made by different people. The beauty of open-source.